luigi/task.py

```diff
@@ -484,6 +484,15 @@ def complete(self):
 
         return all(itertools.imap(lambda output: output.exists(), outputs))
 
+    @classmethod
+    def bulk_complete(cls, parameter_tuples):
+        """Returns those of parameter_tuples for which this Task is complete.
+
+        Override (with an efficient implementation) for efficient scheduling
+        with range tools. Keep the logic consistent with that of complete().
+        """
+        raise NotImplementedError
```

Comment:
1. I share that skepticism, inasmuch as I kept all the magic parts (inference from filesystem listings for the common case) out, in range.py. Still this kind of a building block is necessary to have EIBTI and support output to arbitrary fancy systems, all the while enabling performance.